### ISSUE TITLE
A Question

### ISSUE THREAD

**Issue by MavenFeng** [prabhuomkar/pytorch-cpp#63]
When I use the CPU to train the model, as the number of iterations increases, Loss.Backward () this function is getting slower and slower  ？However, the memory on vs2017 has been stable。
`for (int i = 0; i < num_epochs; i++) {
	
		int batch_index = 0;
		
		for (auto &batch : *dataloader) {
		
			auto data = batch.data.to(device);
			auto target = batch.target.to(device);

			auto output= ae->forward(target);
			
			Tensor LossToTal;
			//Replace torch::nn:functional::mse_loss() is Same   
			s->caclSSIM(data, output, LossToTal);	
			
			auto Loss =1.0f- LossToTal.mean();

			sts = clock();
			optimizer.zero_grad();
			Loss.backward();
			optimizer.step();
			endt = clock();
			v1 = endt - sts;
			if ((batch_index + 1) % 2 == 0) {
				std::cout << "Epoch [" << i << "/" << num_epochs << "], Step [" << batch_index + 1 << "/"
					<< num_samples / batch_size << "], MeanLoss " << Loss.item<double>() << 
					"CostTime:"<<v1<<std::endl;			
			}
			
			batch_index++;
		}
		
		
		
	}`

**Comment by mfl28** [comment]
Hard to say from this code alone.
Just to clarify: When you say slower you mean the value of `v1` gets larger or do the later iterations get noticeably slower (or both)?

Have you tried to measure *wall time* using `std::chrono::high_resolution_clock::now()` as described in the example [here](https://en.cppreference.com/w/cpp/chrono/c/clock) instead of *CPU time* with `std::clock()`?

**Comment by MavenFeng** [comment]
> Hard to say from this code alone.
> Just to clarify: When you say slower you mean the value of `v1` gets larger or do the later iterations get noticeably slower (or both)?
> 
> Have you tried to measure _wall time_ using `std::chrono::high_resolution_clock::now()` as described in the example [here](https://en.cppreference.com/w/cpp/chrono/c/clock) instead of _CPU time_ with `std::clock()`?
thank for your reply,
I used the function you mentioned, but the result is the same
 v1  gets larger  -> loss.backward() cost time gets larger
for example , [epoch 1 ]loss.backward() only cost 200ms every batch   but [epoch 2] cost  500+ms ……  3 ,4 ++ time gets larger  
Do you have a similar problem?

**Comment by mfl28** [comment]
No, I have not seen any slow down in later iterations in any of our tutorials. However, we currently only measure epoch timings in the image captioning tutorial.

Clarifying further, from your comment in the code, you also get this slowing down when just using `torch::nn:functional::mse_loss()` as your loss function?

And out of curiosity: Why are you passing target as the argument and not data in the forward step?

**Comment by MavenFeng** [comment]
> No, I have not seen any slow down in later iterations in any of our tutorials. However, we currently only measure epoch timings in the image captioning tutorial.
> 
> Clarifying further, from your comment in the code, you also get this slowing down when just using `torch::nn:functional::mse_loss()` as your loss function?
> 
> And out of curiosity: Why are you passing target as the argument and not data in the forward step?


1.just using mse_loss   the time is also become slower 
2.hahahahaha because i forget to swap(data,target) so .... target means->data

**Comment by mfl28** [comment]
Then I guess that the cause to your problem must be somewhere in your model-implementation and/or your data-loading.

**Comment by MavenFeng** [comment]
> ere in


thank for your help  i guess same as you  ,
now i need to check my code and compare with your code;

**Comment by mfl28** [comment]
No problem! Closing this for now.
